testing/scenario_app/android/app/src/main/java/dev/flutter/scenarios/TextPlatformViewActivity.java

```diff
@@ -70,6 +74,33 @@ public void configureFlutterEngine(FlutterEngine flutterEngine) {
         .registerViewFactory("scenarios/textPlatformView", new TextPlatformViewFactory());
   }
 
+  @Override
+  public void onFlutterUiDisplayed() {
+    final Intent launchIntent = getIntent();
+    if (!launchIntent.hasExtra("scenario")) {
+      return;
+    }
+    BasicMessageChannel<String> channel =
+        new BasicMessageChannel<>(
+            getFlutterEngine().getDartExecutor(), "set_scenario", StringCodec.INSTANCE);
+    channel.send(launchIntent.getStringExtra("scenario"));
+    notifyFlutterRenderedAfterVsync();
+  }
+
+  private void notifyFlutterRenderedAfterVsync() {
+    // Wait 1s after the next frame, so the Android texture are rendered.
```

Comment:
Let's try this for now. The iOS simulators on LUCI has had some pretty wacky performance.